src/python/ddapp/tabledemo.py

```diff
@@ -1194,6 +1196,14 @@ def onPropertyChanged(self, propertySet, propertyName):
         elif propertyName == 'Scene':
             self.tableDemo.sceneID = self.params.getProperty('Scene')
 
+    def pickupMoreObjects(self):
+        if len(self.tableDemo.clusterObjects) > 0: # There is still sth on the table, let's do it again!
+            print "There are more objects on the table, going at it again!"
+            self.taskTree.selectTaskByName('reach')
```

Comment:
I think this is probably working very well for you.  It's also possible to make a task that adds new tasks to the queue.  The result would be that list of tasks grows each loop, instead of returning to the start, which may or may not be desirable.